include/revng/Model/TupleTree.h

```diff
@@ -1043,22 +1056,32 @@ class TupleTreeReference {
 
   const TupleTreePath &path() const { return Path; }
 
-  T *get() {
-    revng_check(Root != nullptr);
-
-    if (Path.size() == 0)
-      return nullptr;
-
-    return getByPath<T>(Path, *Root);
+  bool hasNullRoot() const {
+    const auto IsNullVisitor = [](const auto &Ptr) { return Ptr == nullptr; };
+    return std::visit(IsNullVisitor, Root);
   }
 
   const T *get() const {
-    revng_check(Root != nullptr);
+    revng_check(Root.index() != std::variant_npos);
+    revng_check(not hasNullRoot());
 
     if (Path.size() == 0)
       return nullptr;
 
-    return getByPath<T>(Path, *Root);
+    const auto GetByPathVisitor =
+      [&Path = std::as_const(Path)](const auto &RootPointer) {
```

Comment:
`std::as_const`, interesting. But is this really necessary or is it just for the sake of it?

Reply:
This was just for the sake of it. Should I drop it?

Reply:
I'd keep it simple and drop it.

Reply:
Done